lib/location/method_gnss.c

```diff
@@ -370,93 +391,153 @@ static void method_gnss_pgps_request_work_fn(struct k_work *item)
 #if defined(CONFIG_NRF_CLOUD_AGPS)
 bool method_gnss_agps_required(struct nrf_modem_gnss_agps_data_frame *request)
 {
-	int type_count = 0;
-
-#if !defined(CONFIG_NRF_CLOUD_PGPS)
-	/* If P-GPS is enabled, use predicted ephemeris to save power instead of requesting them
-	 * using A-GPS.
-	 */
-	if (request->sv_mask_ephe) {
-		type_count++;
-	}
-	if (request->sv_mask_alm) {
-		type_count++;
-	}
-#endif
-	if (request->data_flags & NRF_MODEM_GNSS_AGPS_GPS_UTC_REQUEST) {
-		type_count++;
-	}
-	if (request->data_flags & NRF_MODEM_GNSS_AGPS_KLOBUCHAR_REQUEST) {
-		type_count++;
-	}
-	if (request->data_flags & NRF_MODEM_GNSS_AGPS_NEQUICK_REQUEST) {
```

Comment:
Klobuchar and NeQuick are alternative models for ionospheric corrections. Although NeQuick is present in the GNSS API, we currently only have support for Klobuchar and there are no plans to add support for NeQuick in the future. While NeQuick would allow more accurate corrections, it's much more resource intensive than Klobuchar.

The reason why I removed it here is that it would not have been present in `method_gnss_assistance_request()` where "all" data_flags bits are set, possible causing confusion why NeQuick is missing there.

That said, maybe the GNSS API should have a note saying that NeQuick corrections are not supported by GNSS. Or, I guess we could even remove NeQuick from the API, if we don't plan on adding support for it later.

Reply:
> I guess we could even remove NeQuick from the API, if we don't plan on adding support for it later.

I'd go for this after checking if we have any plans for supporting it

Reply:
I created a Jira ticket for this.